File: problem28/p28.py
def number_spiral_diagonals(n):
    res = 0
    for i in range(3, n+1, 2):
        sq = i ** 2
        res += sq
        r = i-1
        for j in range(1, 4):
            res += sq - (r * j)
    return res + 1


print(number_spiral_diagonals(1001))

File: problem28/p28.java
public class p28 {
	
	public static int numberSpiralDiagonals(int n) {
	    int sum = 0, r, sq;
	    for(int i=3; i<=n; i+=2){
	        sq = i * i;
	        sum += sq;
	        r = i-1;
	        for(int j=1; j<=3; j++){
	            sum += sq - (r * j);
	        }
	    }
	    return sum + 1;
	}
	
	public static void main(String[] args) {
		System.out.println(numberSpiralDiagonals(1001));
	}
	
}
